src/makeInfinityAdder.js

```diff
@@ -4,7 +4,20 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(number) {
+    if (number !== undefined) {
+      sum += number;
+      return adder;
+    } else {
+      const result = sum;
+      sum = 0;
+      return result;
```

Comment:
While the function seems to work correctly for the intended use case, it can be simplified by returning the function directly instead of using an else block. This is because if the `if` condition is met, the function will return early, making the `else` unnecessary. Simplifying the control flow can make the code more readable and maintainable.